views/menus/stav.js

```diff
@@ -4,67 +4,199 @@ import React from 'react'
+      dataSource: new ListView.DataSource({
+        getSectionData: getSectionData,
+        getRowData: getRowData,
+        rowHasChanged: (row1, row2) => row1 !== row2,
+        sectionHeaderHasChanged: (s1, s2) => s1 !== s2,
+      }),
+    }
   }
 
-  componentWillMount() {
-    NetInfo.isConnected.fetch().then(isConnected => {
-      this._onConnectivityChange(isConnected)
-    })
-    NetInfo.isConnected.addEventListener('change', this._onConnectivityChange.bind(this))
+  componentDidMount(){
+    this.fetchData()
   }
 
-  componentWillUnmount() {
-    NetInfo.isConnected.removeEventListener('change', this._onConnectivityChange.bind(this))
+  getTodaysLunchHours(currentDay){
+    for (let i = 0; i < buildingHours.length; i++){
+      if (buildingHours[i].name=='Stav Hall Lunch'){
+        return buildingHours[i].times.hours[currentDay][1]
+      }
+    }
   }
 
-  _onConnectivityChange(isConnected: boolean) {
-    this.setState({isConnected})
+  whichMeal(){
+    let current = moment().tz(CENTRAL_TZ)
+    let currentTime = current.format('H:mm')
+    let lunchClose = this.getTodaysLunchHours(current.format('ddd'))
+
+    if (currentTime < lunchClose){
+      return 1
+    } else {
+      return 2
+    }
   }
 
-  onShouldStartLoadWithRequest() {
-    return true
+  buildStation(station, items){
+    let stationItems = []
+    for (let i = 0; i < station.items.length; i++){
+      let food = items[station.items[i]]
+      if (food.special == 1){
+        let temp = {
+          name: food.label,
+          dietary: food.cor_icon,
+        }
+
+        stationItems.push(temp)
+      }
+    }
+    return stationItems
   }
 
-  onNavigationStateChange(navState: {url: string}) {
-    this.setState({url: navState.url})
+
+  fetchData () {
+    fetch('http://legacy.cafebonappetit.com/api/2/menus?cafe=261').then(response => response.json()).then(responseData => {
+      let items = responseData.items   
+
+      const dataBlob = {}
+      const sectionIDs = []
+      const rowIDs = []
+
+      let meal = responseData.days[0].cafes[261].dayparts[0][this.whichMeal()]
+
+      let stations = meal.stations
+
+      for (let sectionId = 0; sectionId < stations.length; sectionId++){
+        let stationItems = this.buildStation(stations[sectionId], items)
+
+        if (stationItems.length > 0) {
+          sectionIDs.push(sectionId)
+
+          let stationName = stations[sectionId].label
+
+          dataBlob[sectionId] = { station: stationName.charAt(0).toUpperCase() + stationName.slice(1) }
+
+          rowIDs.push([])
+
+          for (let j = 0; j < stationItems.length; j++) {
+
+            const rowId = `${sectionId}:${j}`
+
+            rowIDs[rowIDs.length - 1].push(rowId)
+
+            dataBlob[rowId] = stationItems[j]
+          }
+        }
+      }
+
+      this.setState({
+        dataSource: this.state.dataSource.cloneWithRowsAndSections(dataBlob, sectionIDs, rowIDs),
+        loaded: true,
+      })
+
+    }).done()
```

Comment:
Does the `.done()` method exist on our promises? I thought it had been removed from the standard…